src/Hedgehog/Extras/Test/Base.hs

```diff
@@ -140,6 +141,14 @@ failWithCustom cs mdiff msg = liftTest $ mkTest (Left $ H.Failure (getCaller cs)
 failMessage :: MonadTest m => CallStack -> String -> m a
 failMessage cs = failWithCustom cs Nothing
 
+-- | Invert the behavior of a property: success becomes failure and vice versa.
+expectFailure :: HasCallStack => H.TestT IO m -> Integration ()
+expectFailure prop = GHC.withFrozenCallStack $ do
```

Comment:
It could be useful to be able to investigate the failure and ascertain if that's the expected failure or an accidental one.